test/unit/job_invocation_report_template_test.rb

```diff
@@ -0,0 +1,67 @@
+require 'test_plugin_helper'
+
+class JobReportTemplateTest < ActiveSupport::TestCase
+  class FakeTask < OpenStruct
+    class Jail < Safemode::Jail
+      allow :action_output, :ended_at
+    end
+  end
+
+  context 'job invocation report template' do
+
+    describe 'correct template is available' do
+      it 'settings select includes only report templates with name job_id' do
+        FactoryBot.create(:report_template, name: 'Template 1')
+        correct_template = FactoryBot.create(:report_template, name: 'Job Invocation Report Template')
+        FactoryBot.create(:template_input, template: correct_template, input_type: 'user', name: 'job_id')
+        templates = Setting::RemoteExecution.job_invocation_report_templates_select
+
+        assert_include templates, 'Job Invocation Report Template'
+      end
+    end
+
+    describe 'test correct result for report template' do
+      let(:fake_json_outputs) do
+        [
+          { type: 'stderr', value: "error" },
+          { type: 'stdout', value: "output" },
+          { type: 'stdebug', value: "debug" },
+        ]
+      end
+      let(:fake_task) { FakeTask.new(action_output: { outputs: fake_json_outputs }, ended_at: Time.new(2020, 12, 1, 0, 0, 0)) }
+
+      it 'should process the result of task to template' do
+        job_invocation = FactoryBot.create(:job_invocation, :with_task)
+        JobInvocation.any_instance.expects(:sub_task_for_host).returns(fake_task)
+
+        template_source = File.read(File.expand_path(File.dirname(__FILE__) + "/../../app/views/report_templates/jobs_invocations.erb"))
+        template = ReportTemplate.import_without_save("Job Invocation Report Template", template_source)
+        template.save!
+        input = template.template_inputs.first
+        params = {template_id: template.id, input_values: {input.id.to_s => {value: job_invocation.id.to_s} } }
+        composer = ReportComposer.new(params)
+        time = Time.now
+        result = composer.render
+
+        # parsing the result, csv file originally
+        data = CSV.parse(result, headers: true)
+
+        stderr_row = data[0].to_h
+        stdout_row = data[1].to_h
+        stdebug_row = data[2].to_h
+
+        assert_equal stderr_row['type'], 'stderr'
+        assert_equal stderr_row['message'], 'error'
+        assert_operator Time.parse(stderr_row['ended_at']), :<, time
```

Comment:
My point was to get the assertion to be more exact... this is almost equal to:
```suggestion
        assert_kind_of Time.parse(stderr_row['ended_at']), Time, 'Parsing of time column failed'
```
What would be ok by me, but we don't need a helper variable (`time`) for that :thinking: